src/herder/FetchingEnvelopes.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "herder/EnvelopeItemMap.h"
```

Comment:
It would be a bit less confusing if you were following a PImpl pattern as part of those refactors - this would allow to include implementation classes only from the cpp file (ie: nobody outside should care about EnvelopeItemMap)